lib/kubernetes-deploy/concurrency.rb

```diff
@@ -0,0 +1,18 @@
+# frozen_string_literal: true
+module KubernetesDeploy
+  module Concurrency
+    MAX_THREADS = 8
+
+    def self.split_across_threads(all_work, &block)
+      return if all_work.empty?
+      raise ArgumentError, "Block of work is required" unless block_given?
+
+      slice_size = ((all_work.length + MAX_THREADS - 1) / MAX_THREADS)
```

Comment:
Smart trick slicing the work into N pieces!

Another common pattern would be to create 8 threads and make each of them `pop` from the `Queue` object (which is a thread safe primitive). But I think your approach is even simpler and we should stick with it.